batcher/aligned-sdk/src/sdk/aggregation.rs

```diff
@@ -46,6 +47,33 @@ impl AggregationModeVerificationData {
     }
 }
 
+// We use a newtype wrapper around `[u8; 32]` because Rust's orphan rule
+// prevents implementing a foreign trait (`IsMerkleTreeBackend`) for a foreign type (`[u8; 32]`).
+#[derive(Default, PartialEq, Eq)]
+struct Hash32([u8; 32]);
+
+impl IsMerkleTreeBackend for Hash32 {
+    type Data = Hash32;
+    type Node = [u8; 32];
+
+    fn hash_data(leaf: &Self::Data) -> Self::Node {
+        let mut hasher = Keccak256::new();
+        hasher.update(leaf.0);
+        hasher.finalize().into()
+    }
+
+    fn hash_leaves(leaves: &[Self::Data]) -> Vec<Self::Node> {
+        leaves.iter().map(|l| l.0).collect()
+    }
```

Comment:
This a bit orthodox, hash_leaves doesn't need to be redefined, but if you need it, then it should use hash_data, else hash_data has no purpose. We can re check it tomorrow


```suggestion
    fn hash_leaves(leaves: &[Self::Data]) -> Vec<Self::Node> {
        leaves.iter().map(|l| l.0).collect()
    }
```

Reply:
You are right! Indeed, when testing this in #1915 I had to  introduce this fix. I added it here as well  [5a28bc8](https://github.com/yetanotherco/aligned_layer/pull/1911/commits/5a28bc85176cd3a6ca5b6dfafab15d4505d3bba6).